src/services/alert/index.js

```diff
@@ -0,0 +1,59 @@
+const hooks = require('feathers-hooks-common');
+const cache = require('memory-cache');
+const globals = require('../../../config/globals');
+
+// add Message Provider Adapter here
+const StatusAdapter = require('./adapter/status');
+
+let messages = [];
+// durtion of cache in min
+const duration = 5;
+let success = false;
+
+/**
+ * Building Message Array
+ * @param {Adapter} ProviderAdapter
+ */
+async function addMessageProvider(ProviderAdapter) {
+	const data = await ProviderAdapter.getMessage(globals.SC_THEME);
+	messages = messages.concat(data.messages);
+	return data.success;
+}
+
+/**
+ * Service to get an array of alert messages from added Message Providers (e.g: status.schul-cloud.org)
+ */
+class AlertService {
+	async find() {
+		messages = [];
+		const cachedMessages = cache.get('cachedMessages');
+		if (cachedMessages) {
+			messages = cachedMessages;
+		} else {
+			// add Message Provider here
+			success = await addMessageProvider(new StatusAdapter()); // status.schul-cloud.org
+
+			if (success) { cache.put('cachedMessages', messages, 1000 * 60 * duration); }
```

Comment:
the cache should keep the messages always instead of clear if only this one object is inside. is it possible to keep the successful result and reload & replace it after some time? then if the request fails, you may still respond with the last successful response from the cache. one minute should be enough as timeout